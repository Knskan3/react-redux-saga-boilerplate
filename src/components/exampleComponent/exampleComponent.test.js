import React from 'react';
import Enzyme, { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-15';
import ExampleComponent from './index';

Enzyme.configure({ adapter: new Adapter() });


const generateProps = () => ({
  title: 'Potato on a bike test',
  actionExampleRequest: jest.fn(),
  show: false,
});

describe('<exampleComponent />', () => {
  // Snapshot testing
  test('Snapshot sould be correct', () => {
    const props = generateProps();
    const component = renderer.create(<ExampleComponent {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('Should render title correctly', () => {
    const props = generateProps();
    const wrap = mount(<ExampleComponent {...props} />);
    expect(wrap.find('div h2.title').text()).toEqual(props.title);
  });
});
