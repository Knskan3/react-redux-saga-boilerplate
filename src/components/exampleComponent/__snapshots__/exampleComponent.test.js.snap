// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<exampleComponent /> Snapshot sould be correct 1`] = `
<div
  className="sc-bdVaJa gNeAJc"
>
  <h2
    className="title sc-bwzfXH dUnxak"
  >
    Potato on a bike test
  </h2>
</div>
`;
